### PR TITLE
Move src/Features/WcPayPromotion to /src-internal/Admin/WcPayPromotion

### LINKED ISSUE #8273: Make src/Features/WcPayPromotion/* internal
Move the files to `./Internal` as appropriate

My estimate: 1 hour

### PR DESCRIPTION
**Pull request by moon0326**
Fixes #8273 

This PR moves `src/Features/WcPayPromotion/*` to `/src-internal/Admin/WcPayPromotion`.

- Added `src-internal` directory and assigned `Automattic\WooCommerce\Internal\Admin` namespace to it as per p90Yrv-2Ki#comment-6968

- `WcPayPromotion` classes stopped working after moving to `src/Internal/Admin/WcPayPromotion` since [get_feature_class](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/Features.php#L111) could not find the Init method anymore. I used `class_alias` to fix the issue. This also guarantees backward compatibility.

### Detailed test instructions:

1. Use test instructions from the original [PR](https://github.com/woocommerce/woocommerce-admin/pull/7550)

no changelog

### CODE REVIEW

File: src/Features/Features.php
Comment (becdetat) [inline src/Features/Features.php:44]: typo

File: src/Features/Features.php
Comment (becdetat) [inline src/Features/Features.php:412]: Why have the aliases as a class field? This only ever gets called with the same array, may as well co-locate the array within this function.

File: src/Features/Features.php
Comment (becdetat) [inline src/Features/Features.php:411]: typo in function name

Comment (moon0326) [reply src/Features/Features.php]: Fixed all three in https://github.com/woocommerce/woocommerce-admin/pull/8307/commits/042dac8bcebd26febbb89c9d14163a7e9ec59b82

Thank you for spotting them 🙇‍♂️

File: src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php
Comment (chihsuan) [inline src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php:19]: I think it would be great to update the class name in the comments [here](https://github.com/woocommerce/woocommerce-admin/blob/d52410a9c9a3bbc51337df2ff618a893606062a5/src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php#L3).

Comment (chihsuan) [reply src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php:19]: I'm also curious why do we use `Wc` prefix but have **WC**PayPromotion and **WC**_Payment_Gateway?

**Review by chihsuan — APPROVED** [review]
Test well. Only left some comments about naming. Pre-approving. 👍

File: src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php
Comment (moon0326) [reply src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php:19]: Thank you @chihsuan 

That's a good question. I think it's something we need to discuss and reach an agreement. I don't have a strong preference.

The only reason I renamed it to `Wc` is that the directory and the other class have `Wc` already. It's mixed at the moment.

Let's change it to `WC` since we're moving. I'll make an update.

Comment (moon0326) [reply src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php]: @chihsuan Updated letter case in https://github.com/woocommerce/woocommerce-admin/pull/8307/commits/a03f05d0255b94d66e0c458052a5e26c74f4d4ab and https://github.com/woocommerce/woocommerce-admin/pull/8307/commits/5a20f6278f8e5c7fdc9fb4e00269a5318a3860f3

I also tested the changes on a JN site since my local mac's filesystem isn't case sensitive.

Comment (moon0326) [reply src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php]: FYI, I'm gonna wait until your PRs are ready to merge since those PRs are merging into this PR.

Comment (chihsuan) [reply src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php]: > Updated letter case in https://github.com/woocommerce/woocommerce-admin/commit/a03f05d0255b94d66e0c458052a5e26c74f4d4ab and https://github.com/woocommerce/woocommerce-admin/commit/5a20f6278f8e5c7fdc9fb4e00269a5318a3860f3

> I also tested the changes on a JN site since my local mac's filesystem isn't case sensitive.

@moon0326 Thanks for updating letter case. Looks good. 💯

Comment (chihsuan) [reply src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php]: @moon0326 I think you can merge this PR first. I'll change the merge target for the other PRs.

Comment (moon0326) [reply src-internal/Admin/WcPayPromotion/WcPaymentGatewayPreInstallWCPayPromotion.php]: @chihsuan sounds good. Merged!

**Pull request merged**
